addons/l10n_no_edi/models/account_edi_format.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, _
+from odoo.addons.account_edi_ubl_bis3.models.account_edi_format import COUNTRY_EAS
+
+import base64
+import markupsafe
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    ####################################################
+    # Import
+    ####################################################
+
+    def _is_ubl(self, filename, tree):
+        """ OVERRIDE so that the generic ubl parser does not parse BIS3 any longer.
+        """
+        is_ubl = super()._is_ubl(filename, tree)
+        return is_ubl and not self._is_ehf_3(filename, tree)
+
+    def _is_ehf_3(self, filename, tree):
+        profile_id = tree.find('./{*}ProfileID')
+        nor_tax = tree.xpath("//*[text()='Foretaksregisteret']")
+        return tree.tag == '{urn:oasis:names:specification:ubl:schema:xsd:Invoice-2}Invoice' \
+            and profile_id is not None \
+            and 'peppol' in profile_id.text \
+            and nor_tax is not None
+
+    def _bis3_get_extra_partner_domains(self, tree):
+        if self.code == 'ehf_3':
+            endpoint = tree.find('./{*}AccountingSupplierParty/{*}Party/{*}EndpointID')
+            if endpoint is not None:
+                scheme = endpoint.attrib['schemeID']
+                if scheme == '0192' and endpoint.text:
+                    return [('l10n_no_bronnoysund_number', '=', endpoint.text)]
+        return super()._bis3_get_extra_partner_domains(tree)
+
+    ####################################################
+    # Export
+    ####################################################
+
+    def _get_ehf_3_values(self, invoice):
+        values = super()._get_bis3_values(invoice)
+        for partner_vals in (values['customer_vals'], values['supplier_vals']):
+            partner = partner_vals['partner']
+            if partner.country_code == 'NO':
+                partner_vals.update({
+                    'bis3_endpoint': partner.l10n_no_bronnoysund_number,
+                    'bis3_endpoint_scheme': '0192',
+                })
+
+        return values
+
+    def _export_ehf_3(self, invoice):
+        self.ensure_one()
+        # Create file content.
+        xml_content = markupsafe.Markup("<?xml version='1.0' encoding='UTF-8'?>")
+        xml_content += self.env.ref('l10n_no_edi.export_ehf_3_invoice')._render(self._get_ehf_3_values(invoice))
+        vat = invoice.company_id.partner_id.commercial_partner_id.vat
+        xml_name = 'ehf-%s%s%s.xml' % (vat or '', '-' if vat else '', invoice.name.replace('/', '_'))
+        return self.env['ir.attachment'].create({
+            'name': xml_name,
+            'datas': base64.encodebytes(xml_content),
```

Comment:
Or set the `raw` attribute and don't need to unnecessarily b64encode/b64decode. That probably requires explicitely `encode`-ing `xml_content`.